### PR TITLE
throw a cpk warning when the feature flag is disabled and datastore is used in an Amplify Project

### PR DESCRIPTION
**Pull request by manaswi223**
#### Checklist

- [ ] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dpilch** [comment]
Looks good, can you add a test to cover the warning message case?

File: packages/amplify-graphql-transformer-core/src/transformation/transform.ts [hunk 1]
```diff
@@ -90,7 +92,7 @@ export interface GraphQLTransformOptions {
   readonly overrideConfig?: OverrideConfig;
 }
 export type StackMapping = { [resourceId: string]: string };
-export class GraphQLTransform {
+export class GraphQLTransform {
```
Comment (david-mcafee) [inline packages/amplify-graphql-transformer-core/src/transformation/transform.ts:95]: I'm not a fan of committing linting changes alongside actual code changes. I would remove the linting updates from this PR.

File: packages/amplify-graphql-transformer-core/src/transformation/transform.ts [hunk 2]
```diff
@@ -822,4 +825,12 @@ export class GraphQLTransform {
       index++;
     }
   }
+
+// Checks if DataStore is enabled and CPK Feature Flag is true. Throws a warning otherwise.
+public validateCPKFeatureFlag = (context: TransformerContext) => {
+  const isDataStoreEnabled = context.isProjectUsingDataStore();
+  if (isDataStoreEnabled && !isCPKFeatureEnabled(context)) {
+    console?.warn(`WARNING: Your schema has a custom primary key but the Feature Flag "${cpkFeatureFlagName}" is disabled. Check the value in your "amplify/cli.json" file, change it to "true" and re-run.`);
```
Comment (david-mcafee) [inline packages/amplify-graphql-transformer-core/src/transformation/transform.ts:833]: DataStore being enabled without the CPK feature flag doesn't mean the schema is using a custom primary key. We should only throw the error if the schema actually contains a custom primary key.

**Review by david-mcafee — CHANGES_REQUESTED** [review]

File: packages/amplify-graphql-transformer-core/src/transformation/transform.ts (hunk 1 above)
Comment (manaswi223) [reply packages/amplify-graphql-transformer-core/src/transformation/transform.ts:95]: sure, can do that!

Comment (david-mcafee) [reply packages/amplify-graphql-transformer-core/src/transformation/transform.ts:95]: Thanks @manaswi223!

File: packages/amplify-graphql-transformer-core/src/transformation/transform.ts (hunk 2 above)
Comment (manaswi223) [reply packages/amplify-graphql-transformer-core/src/transformation/transform.ts:833]: oh wow, good catch David! Will change accordingly 🤦🏼‍♀️

Comment (manaswi223) [reply packages/amplify-graphql-transformer-core/src/transformation/transform.ts]: updated with validating hasprimarykey() or not :)

File: packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts [hunk 3]
```diff
@@ -116,10 +119,8 @@ function validate(config: PrimaryKeyDirectiveConfiguration, ctx: TransformerCont
   const { object, field, sortKeyFields } = config;
 
   validateNotSelfReferencing(config);
-
-  const modelDirective = object.directives!.find(directive => {
-    return directive.name.value === 'model';
-  });
+  validateCPKFeatureFlag(ctx,config);
```
Comment (phani-srikar) [inline packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts:122]: If there are 2 models with primary keys, can you confirm that we don't throw the warning twice?

File: packages/amplify-graphql-transformer-core/src/transformation/transform.ts [hunk 4]
```diff
@@ -1,825 +0,0 @@
-/* eslint-disable no-new */
```
Comment (phani-srikar) [inline packages/amplify-graphql-transformer-core/src/transformation/transform.ts:1]: I don't think you meant to delete these files. Please re-add them.

**Review by phani-srikar — CHANGES_REQUESTED** [review]

File: packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts (hunk 3 above)
Comment (manaswi223) [reply packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts]: Thank you Phani - it was warning 2 times when my schema has 2 @primarykey.
So, changed my code and now the warning isn't producing twice

**Comment by david-mcafee** [comment]
@manaswi223 - is [https://github.com/aws-amplify/amplify-category-api/pull/1133](https://github.com/aws-amplify/amplify-category-api/pull/1133) the latest PR for this change? If so, can we close this PR to eliminate confusion? Thanks!

**Pull request closed without merging**